dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/MetadataQueryGenerator.java

```diff
@@ -110,21 +111,38 @@ static String createSelectForTableTextV(
         .generate();
   }
 
+  static String createSelectForDiskSpaceV(OptionalLong rowCount, Optional<Expression> condition) {
+    return select("%s")
+        .from(subquery(createLimitedSelect(rowCount, "DBC.DiskSpaceV", condition, "CurrentPerm")))
+        .as("t")
+        .serialize();
+  }
+
   private static SelectExpression createSingleDbKindSelectFromDatabasesV(
       String dbKind, OptionalLong rowCount, Optional<Expression> condition) {
-    String tableName = "DBC.DatabasesV";
     Expression dbKindCondition = eq(identifier("DBKind"), stringLiteral(dbKind));
-    Expression processedCondition =
-        condition
-            .<Expression>map(innerCondition -> and(innerCondition, dbKindCondition))
-            .orElse(dbKindCondition);
-    if (rowCount.isPresent()) {
-      return selectTop(rowCount.getAsLong(), projection(star()))
-          .from(tableName)
-          .where(processedCondition)
-          .orderBy(OrderBySpec.create(identifier("PermSpace"), Direction.DESC));
-    } else {
-      return selectAll().from(tableName).where(processedCondition).build();
-    }
+    return createLimitedSelect(
+        rowCount,
+        "DBC.DatabasesV",
+        Optional.of(
+            condition
+                .<Expression>map(innerCondition -> and(innerCondition, dbKindCondition))
+                .orElse(dbKindCondition)),
+        "PermSpace");
+  }
+
+  private static SelectExpression createLimitedSelect(
+      OptionalLong rowCount,
+      String tableName,
+      Optional<Expression> condition,
+      String orderByColumnDesc) {
+    FromClauseStepBuilder builder =
+        rowCount.isPresent()
+            ? selectTop(rowCount.getAsLong(), projection(star())).from(tableName)
+            : selectAll().from(tableName);
+    AfterWhereStepBuilder builder2 = condition.map(builder::where).orElse(builder);
+    return rowCount.isPresent()
+        ? builder2.orderBy(OrderBySpec.create(identifier(orderByColumnDesc), Direction.DESC))
```

Comment:
Why orderBy is used if rowCount is provided only? I guess order doesn't matter when we do not limit number of returned records, but this logic is misleading. rowCount is OptionalLong and orderByColumnDesc is a String so I would expect I can provide order column with no rowCount.

Reply:
or if it is an expected behavior can we add test case for it?